builtin/providers/aws/resource_aws_key_pair.go

```diff
@@ -65,6 +65,11 @@ func resourceAwsKeyPairRead(d *schema.ResourceData, meta interface{}) error {
 	}
 	resp, err := conn.DescribeKeyPairs(req)
 	if err != nil {
+		awsErr, ok := err.(aws.APIError)
+		if ok && awsErr.Code == "InvalidKeyPair.NotFound" {
```

Comment:
I don't see why it would merit a warning. As I see it, this is just one among many ways the AWS state could diverge from the desired state as described by the configuration. If I'm concerned about monitoring unintended deviations of state, I can just run `terraform plan` (having not changed the configuration) and see all the deviations.